k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolderPusher.java

```diff
@@ -504,6 +503,12 @@ private int processUntaggedResponse(long oldMessageCount, ImapResponse response,
                             Timber.d("Got untagged EXPUNGE for msgseq %d for %s", msgSeq, getLogId());
                         }
 
+                        if (K9MailLib.isDebug()) {
+                            Timber.d("Scheduling removal of message with seq %d due to expunge", msgSeq);
+                        }
+
+                        removeMsgUids.add("thisisahack");
```

Comment:
Whoops, I was going to add a comment about this, but it slopped my mind. This code does not trigger the removal of messages from the local store when an EXPUNGE message is received through IDLE. This is because EXPUNGE responses contain only the sequence number of the corresponding messages, and we depend on a mapping of sequence numbers to UIDs to get the corresponding UID. This mapping is populated through previously received FETCH responses. When CONDSTORE/QRESYNC is used, we send FETCH requests for only a selected subset of messages (the ones which have changed), so the UIDs of the remaining ones are not available and cannot be expunged from the local store.

I suppose the simplest way to fix it would be to use a UID SEARCH command to get the corresponding UID. However, since my next task is to rewrite the IDLE-handling code, which will invoIve removing most of this, I decided to do that instead.